code/modules/tgui/tgui.dm

```diff
@@ -142,6 +149,9 @@
 	src_object.on_ui_open(user, src)
 	for(var/datum/module as anything in modules_registered)
 		module.on_ui_open(user, src, TRUE)
+	// now send deferred asets
+	for(var/datum/asset_pack/assetlike in assets_deferred)
+		window.send_asset(assetlike)
```

Comment:
window.send_asset is blocking

this should probably be addtimer/spawn at some point but i'm lazy + for now it's fine
i also haven't analyzed the impact of making this non-blocking on the large scale.